### PR TITLE
[hailtop.batch] expose batch pool executor docs

### PR DESCRIPTION
**Pull request by danking**
Do I have to do anything else to get BatchPoolExecutor exposed in the Batch docs?

### CODE REVIEW

File: hail/python/hailtop/batch/docs/api.rst [hunk 1]
```diff
@@ -58,6 +58,22 @@ share a common root, but each file has its own extension.
...
+A :class:`.BatchPoolExecutor` provides roughly the same interface as the Python
+standard library's :class:`.concurrent.futures.Executor`. It facilitates
+executing arbitrary python functions in the cloud.
+
+.. autosummary::
+    :toctree: api/batch/
+    :nosignatures:
+    :template: class.rst
+
+    batch.Batch
```
Comment (jigold) [inline hail/python/hailtop/batch/docs/api.rst:73]: This isn't the right thing. I think you need batch_pool_executor.BatchPoolExecutor.

File: hail/python/hailtop/batch/docs/api.rst [hunk 2]
```diff
@@ -58,6 +58,22 @@ share a common root, but each file has its own extension.
...
 
+Batch Pool Executor
+~~~~~~~~~~~~~~~~~~~
+
+A :class:`.BatchPoolExecutor` provides roughly the same interface as the Python
+standard library's :class:`.concurrent.futures.Executor`. It facilitates
+executing arbitrary python functions in the cloud.
+
+.. autosummary::
+    :toctree: api/batch/
```
Comment (jigold) [inline hail/python/hailtop/batch/docs/api.rst:69]: Make this api/batch_pool_executor/. This is where the rst docs will be generated to.

**Comment by danking** [comment]
Looks good locally to me now

File: hail/python/hailtop/batch/batch_pool_executor.py [hunk 3]
```diff
@@ -373,16 +376,28 @@ def _finish_future(self):
             self._cleanup(False)
 
     def shutdown(self, wait=True):
+        """Allow temporary resources to be cleaned up.
+
+        Until shutdown is called, some temporary cloud storage files will
+        persist. These files will after both shutdown has been called and all
```
Comment (jigold) [inline hail/python/hailtop/batch/batch_pool_executor.py:382]: I think you want `will be deleted`.

File: hail/python/hailtop/batch/docs/api.rst [hunk 4]
```diff
@@ -58,6 +58,21 @@ share a common root, but each file has its own extension.
     resource.ResourceGroup
 
 
+Batch Pool Executor
+~~~~~~~~~~~~~~~~~~~
+
+A :class:`.BatchPoolExecutor` provides roughly the same interface as the Python
+standard library's :class:`.concurrent.futures.Executor`. It facilitates
+executing arbitrary python functions in the cloud.
```
Comment (jigold) [inline hail/python/hailtop/batch/docs/api.rst:66]: Are we capitalizing Python?

**Pull request merged**